hydro_cli/src/core/progress.rs

```diff
@@ -264,9 +264,9 @@ impl ProgressTracker {
         }
     }
 
-    pub fn leaf_with_progress<'a, T, F: Future<Output = T>>(
+    pub fn rich_leaf<'a, T, F: Future<Output = T>>(
         name: String,
-        f: impl FnOnce(Arc<dyn Fn(u64) + Send + Sync>) -> F + 'a,
+        f: impl FnOnce(Arc<dyn Fn(u64) + Send + Sync>, Arc<dyn Fn(String) + Send + Sync>) -> F + 'a,
```

Comment:
Looking at this, these really should be `Box<dyn Fn(...) + ...>`, doesnt actually use `Arc`